File: SDKs/amazonaws.com/elasticache/2015-02-02/java/lib/src/main/java/openapisdk/models/shared/DestinationDetails.java
package openapisdk.models.shared;



/**
 * DestinationDetails
 * Configuration details of either a CloudWatch Logs destination or Kinesis Data Firehose destination.
**/
public class DestinationDetails {
    public CloudWatchLogsDestinationDetails cloudWatchLogsDetails;
    public DestinationDetails withCloudWatchLogsDetails(CloudWatchLogsDestinationDetails cloudWatchLogsDetails) {
        this.cloudWatchLogsDetails = cloudWatchLogsDetails;
        return this;
    }
    public KinesisFirehoseDestinationDetails kinesisFirehoseDetails;
    public DestinationDetails withKinesisFirehoseDetails(KinesisFirehoseDestinationDetails kinesisFirehoseDetails) {
        this.kinesisFirehoseDetails = kinesisFirehoseDetails;
        return this;
    }
}

File: SDKs/amazonaws.com/elasticache/2015-02-02/python/src/sdk/models/shared/destinationdetails.py
import dataclasses
from typing import Optional
from ..shared import cloudwatchlogsdestinationdetails as shared_cloudwatchlogsdestinationdetails
from ..shared import kinesisfirehosedestinationdetails as shared_kinesisfirehosedestinationdetails


@dataclasses.dataclass
class DestinationDetails:
    r"""DestinationDetails
    Configuration details of either a CloudWatch Logs destination or Kinesis Data Firehose destination.
    """
    
    cloud_watch_logs_details: Optional[shared_cloudwatchlogsdestinationdetails.CloudWatchLogsDestinationDetails] = dataclasses.field(default=None)
    kinesis_firehose_details: Optional[shared_kinesisfirehosedestinationdetails.KinesisFirehoseDestinationDetails] = dataclasses.field(default=None)
    
